client/wfnews-war/src/main/angular/src/app/components/current-alert-component/current-alert-component.component.ts

```diff
@@ -1,41 +1,89 @@
   public evacOrders : EvacOrderOption[] = []
   public areaRestrictions: AreaRestrictionsOption[] = []
   public bansAndProhibitions: BansAndProhibitionsOption[] = []
-  currentLat: number;
-  currentLong: number;
-  nearmeChecked: boolean;
-  alertTypeOptions = [
+  public currentLat: number;
+  public currentLong: number;
+  public nearmeChecked: boolean;
+  public alertTypeOptions = [
     'All',
     'Area Restrictions',
     'Bans & Prohibitions',
     'Evacuation Orders and Alerts',
   ]
-  selectedAlertType = 'All'
+  public selectedAlertType = 'All'
 
-  constructor(private agolService: AGOLService, private commonUtilityService: CommonUtilityService) {
-}
+  constructor(private agolService: AGOLService, private commonUtilityService: CommonUtilityService, private appConfigService: AppConfigService) {}
   public convertToDateWithDayOfWeek = DateTimeConvert;
 
-
-
   ngOnInit() {
     this.useMyCurrentLocation()
-    this.getEvacOrders(null, null);
-    this.getBansProhibitions(null, null);
-    this.getAreaRestrictions(null, null)
+  }
+
+  createMap (id: string, item: any, type: string) {
+    const location = item && item.centroid ? [Number(item.centroid.y || 0), Number(item.centroid.x || 0)] : [0, 0]
+    const map = L.map(id, {
+      attributionControl: false,
+      zoomControl: false,
+      dragging: false,
+      doubleClickZoom: false,
+      boxZoom: false,
+      trackResize: false
+    }).setView(location, 12)
+
+    if (item.geometry) {
+      map.fitBounds(new L.LatLngBounds([item.geometry.ymin, item.geometry.xmin], [item.geometry.ymax, item.geometry.xmax]));
+    }
+
+    L.tileLayer('https://tile.openstreetmap.org/{z}/{x}/{y}.png', {
+      attribution: '&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
+    }).addTo(map)
+
+    let databcUrl = this.appConfigService.getConfig()['mapServices']['openmapsBaseUrl'].toString()
+
+    if (type.toLocaleLowerCase() === 'evac') {
+      L.tileLayer.wms(databcUrl, {
+        layers: 'WHSE_HUMAN_CULTURAL_ECONOMIC.EMRG_ORDER_AND_ALERT_AREAS_SP',
+        styles: '6885',
+        format: 'image/png',
+        transparent: true,
+        version: '1.1.1'
+      }).addTo(map);
+    } else if (type.toLocaleLowerCase() === 'ban') {
+      L.tileLayer.wms(databcUrl, {
+        layers: 'WHSE_LAND_AND_NATURAL_RESOURCE.PROT_BANS_AND_PROHIBITIONS_SP',
+        styles: '7733',
+        format: 'image/png',
+        transparent: true,
+        version: '1.1.1'
+      }).addTo(map);
+    } else if (type.toLocaleLowerCase() === 'area') {
+      L.tileLayer.wms(databcUrl, {
+        layers: 'WHSE_LAND_AND_NATURAL_RESOURCE.PROT_RESTRICTED_AREAS_SP',
+        styles: '7735',
+        format: 'image/png',
+        transparent: true,
+        version: '1.1.1'
+      }).addTo(map);
+    }
+  }
+
+  async ngAfterViewInit () {
+    await Promise.all([this.getEvacOrders(), this.getBansProhibitions(), this.getAreaRestrictions()])
```

Comment:
Generally because we want to tie them to the map, which renders as a post process after the view is created. These promises are asyn/indeterminate, so the ViewInit will not trigger the AfterViewInit otherwise.

You can move those promises out into the ViewInit, and the maps will no longer render

Reply:
Basically, the Angular lifecycle can be weird sometimes